terraform-plans/configs/charm-advanced-routing_main.tfvars

```
@@ -51,4 +51,9 @@ templates = {
       repository = "charm-advanced-routing"
     }
   }
+  pyproject = {
+    source      = "./templates/github/pyproject.toml.tftpl"
+    destination = "pyproject.tom"
```

I think you are missing a `l` in those file names

```suggestion
    destination = "pyproject.toml"
```